src/infra/core/src/repos/dataset_impl.rs

```diff
@@ -356,19 +357,29 @@ where
         let output_data = match data_interval {
             None => None,
             Some(data_interval) => {
-                let from_data_path = movable_data_file.unwrap();
+                let from_data_path = Arc::new(movable_data_file.unwrap().to_path_buf());
+
+                let path = from_data_path.clone();
+                let logical_hash = tokio::task::spawn_blocking(move || {
```

Comment:
Just by looking at the trace I was able to easily spot that main thread is blocked for a very long time when hashing - so adding a fix that moves hashing to a background thread.

Reply:
I was actually recently surprised those data hashing helper functions were synchronous, when I did `QueryService` refactoring.